src/disk.hpp

```diff
@@ -100,20 +100,35 @@ struct FileDisk {
         Open(false);
     }
 
-    void Open(bool read_only = true)
+    void Open(bool read_only = true, bool retry = false)
```

Comment:
This isn't directly an issue with this patch, but I think it's problematic to take multiple plain `bool`s like this. At the call site it will be very easy to misunderstand what each `true` or `false` mean.

I would suggest we instead pass a flag here, had have two named flags `read_only` and `retry`. That would improve the current situation with regards to readability at the call site as well.

Reply:
Sure, I can do that. I'm bad at C++ though, so I'd appreciate some pointers on where you want such flags to be defined. Maybe a nested struct in `FileDisk`?

```suggestion
    // Define right under `struct FileDisk {` as nested struct? Top-level in this file?
    struct FileOpenFlags {
        bool read_only = true;
        bool retry = false;
    }
    
    void Open(FileOpenFlags options)
```

Reply:
Or maybe you meant something more like passing `uint8_t flags` into `Open`?